src/document-submit.js

```diff
@@ -0,0 +1,111 @@
+/**
```

Comment:
See my comment with respect to whether it should be a service.

Adding custom rules to dep-check-config is easy :)